nixos/modules/services/networking/nebula.nix

```diff
@@ -229,7 +252,7 @@ in
 
     # Open the chosen ports for UDP.
     networking.firewall.allowedUDPPorts =
-      unique (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks);
+      unique (filter (port: port != null && port > 0) (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks));
```

Comment:
This doesn't apply unless the port is explicitly set (i.e. lighthouses are implicitly assigned port 4242, but that port is not opened).

Reply:
Other than this, tested and working on my own network!

Reply:
Thanks! I'll fix that. Thanks for the double check.

Reply:
I think something like this would work to avoid duplicating logic:
```diff
diff --git a/nixos/modules/services/networking/nebula.nix b/nixos/modules/services/networking/nebula.nix
index de68f514f5a6..40ef6f952ad5 100644
--- a/nixos/modules/services/networking/nebula.nix
+++ b/nixos/modules/services/networking/nebula.nix
@@ -10,6 +10,15 @@ let
   format = pkgs.formats.yaml {};
 
   nameToId = netName: "nebula-${netName}";
+
+  resolveFinalPort = netCfg:
+    if netCfg.listen.port == null then
+      if (netCfg.isLighthouse || netCfg.isRelay) then
+        4242
+      else
+        0
+    else
+      netCfg.listen.port;
 in
 {
   # Interface
@@ -181,15 +190,7 @@ in
           };
           listen = {
             host = netCfg.listen.host;
-            port = (
-              if netCfg.listen.port == null then
-                if (netCfg.isLighthouse || netCfg.isRelay) then
-                  4242
-                else
-                  0
-              else
-                netCfg.listen.port
-            );
+            port = resolveFinalPort netCfg;
           };
           tun = {
             disabled = netCfg.tun.disable;
@@ -252,7 +253,7 @@ in
 
     # Open the chosen ports for UDP.
     networking.firewall.allowedUDPPorts =
-      unique (filter (port: port != null && port > 0) (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks));
+      unique (filter (port: port > 0) (mapAttrsToList (netName: netCfg: resolveFinalPort netCfg) enabledNetworks));
 
     # Create the service users and groups.
     users.users = mkMerge (mapAttrsToList (netName: netCfg:
```